### ISSUE TITLE
Allow a pattern to define its own type and subtype

### ISSUE THREAD

**Issue by dmolsen** [pattern-lab/patternlab-php-core#34]
Allow a pattern to define properties about itself via a JSON schema. At a basic level this would include type and subtype. Not sure how practical this really is unfortunately.

**Comment by aleksip** [comment]
These could also be defined in the PL `extra` section in `composer.json`. Type and subtype could even make up the `defaultPath` in #30?

**Comment by dmolsen** [comment]
Good call. I'll have to think that through a little more but it makes sense on the surface.
